src/lib/components/ContactForm.svelte

```diff
@@ -38,7 +38,14 @@
 				maxlength={MAX_NAME_LIMIT}
 				name="from"
 				placeholder="From"
-				value={contactForm.prefillWithSession ? sessionEmail ?? '' : ''}
+				value={contactForm.prefillWithSession && $page.data.contactModes.includes('email')
+					? sessionEmail ?? ''
+					: ''}
+				pattern={$page.data.contactModes.includes('email')
+					? ''
+					: $page.data.contactModes.includes('nostr')
+					? '^(?!.*@).*'
+					: '^(?!npub).*'}
```

Comment:
For emails, instead of a pattern, you can add `type=email` instead of `type=text`